lookout/style/typos/correction_model.py

```diff
@@ -0,0 +1,408 @@
+                numpy.sum(labels[:edge]))
+
+        evallist = [(dtrain, 'train'), (dval, 'validation')]
+        self.bst = xgb.train(self.bst_param,
+                             dtrain,
+                             self.TRAIN_ROUND_NUM,
+                             evallist,
+                             early_stopping_rounds=self.EARLY_STOPPING_ROUNDS)
+
+    def suggest_file(self, test_file, cand_test_file=None, n_candidates=None,
+                     return_all=True, start_pool_size=100) -> dict:
+        test_df = pandas.read_pickle(test_file)
+        save_cand_file = None
+        if cand_test_file is None:
+            path_split = path.split(test_file)
+            cand_test_file = path.join(path_split[0],
+                                       "v1_cand_" + path_split[1])
+        return self.suggest_df(test_df,
+                               cand_test_file=cand_test_file,
+                               save_cand_file=save_cand_file,
+                               n_candidates=n_candidates,
+                               return_all=return_all,
+                               start_pool_size=start_pool_size)
+
+    def suggest_by_batches(self, test_df, n_candidates=None,
+                           return_all=True, batch_size=2048):
+        test_df = add_context_info(test_df)
+
+        all_suggestions = []
+        for i in tqdm(range(0, len(test_df), batch_size)):
+            all_suggestions.append(self.suggest_df(
+                test_df.loc[test_df.index[i]:
+                            test_df.index[min(len(test_df) - 1,
+                                              i + batch_size - 1)], :],
+                n_candidates=n_candidates,
+                return_all=return_all))
+
+        suggestions = dict(chain.from_iterable(
+            [suggs.items() for suggs in all_suggestions]))
+        return suggestions
+
+    def suggest_df(self, test_df, cand_test_file=None, save_cand_file=None,
+                   n_candidates=None, return_all=True, start_pool_size=100):
+        test_df = add_context_info(test_df)
+
+        if cand_test_file is not None:
+            candidates = pandas.read_pickle(cand_test_file)
+            path_split = path.split(cand_test_file)
+            matrix = numpy.load(path.join(path_split[0],
+                                          "matrix_" + path_split[1]))
+        else:
+            if save_cand_file is None:
+                save_cand_file = "cand_test.pkl"
+            candidates, matrix = self._create_candidates(
+                test_df,
+                save_cand_file=save_cand_file,
+                start_pool_size=start_pool_size)
+
+        dtest = xgb.DMatrix(matrix)
+        test_proba = self.bst.predict(dtest,
+                                      ntree_limit=self.bst.best_ntree_limit)
+
+        return suggest_corrections(candidates, test_proba,
+                                   n_candidates, return_all)
+
+    class TypoInfo:
```

Comment:
this class should have its attributes in it (id, typo etc). See typing.NamedTuple